SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialConnectionManager.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/Connection/SpatialOSWorkerInterface.h"
+#include "Interop/Connection/ConnectionConfig.h"
+#include "SpatialCommonTypes.h"
+#include "SpatialGDKSettings.h"
+
+#include "SpatialConnectionManager.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialConnectionManager, Log, All);
+
+class USpatialWorkerConnection;
+
+enum class ESpatialConnectionType
+{
+	Receptionist,
+	LegacyLocator,
+	Locator,
+	DevAuthFlow
+};
+
+UCLASS()
+class SPATIALGDK_API USpatialConnectionManager : public UObject
```

Comment:
Possibly out of scope for this PR, as this is a reorganisation of existing code rather than a rewrite. If so feel free to resolve and the discussion can happen offline.

As the connection manager is meant to create the connection, does it need to be a UObject and is it right that it should own the connection?

It looks like this only reason this is a UObject is because the worker connection is a UObject which I'm not sure needs to be either.

Could it instead be some object we build up some function that returns what is effectively a future. We check the future each tick then eventually give the resulting connection to the workerconnection object, or retry if it fails (which I think we don't currently have any ability to do anyway.

Reply:
I agree that it doesn't need to be a UObject, but I explicitly didn't want to do that in this PR because I wanted to be able to isolate ownership/memory management issues with the changing objects because of the restructuring, as opposed to as a result of a UObject conversion. I'd be very interested in doing that as a follow-up.